src/sentry/models/event.py

```diff
@@ -83,6 +83,17 @@ def _get_project(self):
 
     project = property(_get_project, _set_project)
 
+    @property
+    def environment(self):
+        from sentry.models import Environment
+
+        if not hasattr(self, '_environment_cache'):
+            self._environment_cache = Environment.get_or_create(
+                project=self.project,
+                name=self.get_tag('environment'),
+            )
+        return self._environment_cache
```

Comment:
Not in super love with this method. I assume we will eventually add an `environment_id` column to `Event`? Until then I need to know the `environment_id` per `Event` for this: https://github.com/getsentry/sentry/pull/6600/files#diff-f167bcbd00dbad28dc1a7a56da7eaf5fR244

Reply:
This also hits Riak, which makes me a little nervous.

In my opinion, abstracting this away as a property access makes this even more unclear. I think it'd be worth considering making this a `get_environment` method instead. To me, that at least makes it obvious that it's doing something more complex than just returning a value.

Reply:
Good call. And honestly, I think for now this should just be a helper in the unmerge code.